src/Servers/Kestrel/Transport.NamedPipes/test/WebHostTests.cs

```diff
@@ -0,0 +1,333 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO.Pipes;
+using System.Net;
+using System.Net.Http;
+using System.Net.Security;
+using System.Security.AccessControl;
+using System.Security.Principal;
+using System.Text;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Connections.Features;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Internal;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.AspNetCore.Server.Kestrel.Transport.NamedPipes.Internal;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.NamedPipes.Tests;
+
+public class WebHostTests : LoggedTest
+{
+    [Fact]
+    public async Task ListenNamedPipeEndpoint_HelloWorld_ClientSuccess()
+    {
+        // Arrange
+        using var httpEventSource = new HttpEventSourceListener(LoggerFactory);
+        var pipeName = NamedPipeTestHelpers.GetUniquePipeName();
+
+        var builder = new HostBuilder()
+            .ConfigureWebHost(webHostBuilder =>
+            {
+                webHostBuilder.UseNamedPipes();
+                webHostBuilder
+                    .UseKestrel(o =>
+                    {
+                        o.Listen(new NamedPipeEndPoint(pipeName));
```

Comment:
Will there be new ListenPipe extensions? How about config support, or IServerAddresses?

Reply:
This project can't add an extension method on KestrelServerOptions because of layering.

Reply:
Yes, we briefly discussed multiple transports and what it would take to make it work. I think it would be worth trying to make it work again.

Reply:
I have some ideas. I'll create an issue to discuss further.

Reply:
Maybe a generic listener that gets specific endpoint types where Kestrel does the selection based on that.